tasks/prime.py

```diff
@@ -879,6 +903,97 @@ def update_post_counseling_information(self, overrides=None):
+        url, request_kwargs = self.request_preparer.prep_prime_request(endpoint="/moves")
+
+        if is_local(env=self.env):
+            request_kwargs["timeout"] = 15  # set a timeout of 15sec if we're running locally for this endpoint
+
+        with self.rest(method="GET", url=url, **request_kwargs) as resp:
+            resp: RestResponseContextManager
+
+            log_response_info(response=resp)
+
+            if resp.status_code == HTTPStatus.OK:
+                moves: JSONArray = resp.js
+            else:
+                resp.failure("Unable to get moves available to prime.")
+
+                log_response_failure(response=resp)
+
+                return
+
+        move = random.choice(moves)
+
+        # Now we need to retrieve the shipments for this move
+        url, request_kwargs = self.request_preparer.prep_prime_request(
+            endpoint=f"/move-task-orders/{move['id']}",
+            endpoint_name="/move-task-orders/{moveID}",
+        )
+
+        with self.rest(method="GET", url=url, **request_kwargs) as resp:
+            resp: RestResponseContextManager
+
+            log_response_info(response=resp)
+
+            if resp.status_code == HTTPStatus.OK:
+                move: JSONObject = resp.js
+            else:
+                resp.failure("Unable to get the move.")
+
+                log_response_failure(response=resp)
+
+                return
+
+        # Since we're going to try setting an invalid status that the shipments we recieved above
+        # can't be in, we'll just grab a random one.
+        mto_shipment = deepcopy(random.choice(move["mtoShipments"]))
+
+        overrides = {"status": "DRAFT"}
+
+        # Generate fake payload based on the endpoint's required fields
+        payload = fake_data_generator.generate_fake_request_data(
+            api_key=APIKey.PRIME,
+            path="/mto-shipments/{mtoShipmentID}/status",
+            method="patch",
+            overrides=overrides,
+        )
+
+        # Note that we have an em dash plus "expected failure" to put these in a separate locust group
+        # than the regular shipment status updates.
+        url, request_kwargs = self.request_preparer.prep_prime_request(
+            endpoint=f"/mto-shipments/{mto_shipment['id']}/status",
+            endpoint_name="/mto-shipments/{mtoShipmentID}/status — expected failure",
+        )
+
+        request_kwargs["headers"]["If-Match"] = mto_shipment["eTag"]
+
+        with self.rest(method="PATCH", url=url, data=json.dumps(payload), **request_kwargs) as resp:
+            resp: RestResponseContextManager
+
+            log_response_info(response=resp)
+
+            if resp.status_code == HTTPStatus.UNPROCESSABLE_ENTITY:
+                # Note that by default, locust would fail this request because its status code is >
+                # 400 so
+                # we need to explicitly mark it as a failure.
+                resp.success()
```

Comment:
I don't know if I was consistent in categorizing expected failures as successes, did you happen to make them consistent in your previous PR?

Do we already have another expected failure test for this endpoint, isn't DRAFT one of the status enum options that might be generated when making a generate_fake_request_data call already?

Reply:
Yeah, I fixed them all in a previous pr to count them as successes.

We have one for the support version of the endpoint, but not for the prime.

The enum excludes DRAFT: https://github.com/transcom/mymove/blob/5420f3d1a9c2b57da1c5530c3b3713fc726ad95e/swagger/prime.yaml#L2494-L2503

The support api has more statuses in the enum, but also excludes draft: https://github.com/transcom/mymove/blob/5420f3d1a9c2b57da1c5530c3b3713fc726ad95e/swagger/support.yaml#L1859-L1869

Reply:
True 🤔 I guess in general the expected failure tests might be better to keep at a lower weight than the rest.

Though I would also say that these should be more like black-box tests in that they shouldn't know the internals of how the api validates the info, just that it gets an expected failure. Like this test should work the same whether we keep using what we are using or if we swapped out our API layer to use a different library. The swagger itself only says what the valid values should be, it's the code that the go swagger package (I think `go-openapi/swag`?) generates that does the validation for us.